pkg/logql/downstream.go

```diff
@@ -71,7 +72,8 @@ func (ng *DownstreamEngine) Query(p Params, mapped syntax.Expr) Query {
 		parse: func(_ context.Context, _ string) (syntax.Expr, error) {
 			return mapped, nil
 		},
-		limits: ng.limits,
+		limits:   ng.limits,
+		resolver: tenant.NewMultiResolver(),
```

Comment:
We cannot simply use `tenant.TenantIDs` because this will use the single tenant resolver unless we override it. This could be fine but I'm wondering if it could break anything.

Reply:
Io order to enable any multi tenant features the global resolver should be swapped to a multi tenanted one. See what Cortex is doing there:

https://github.com/cortexproject/cortex/blob/45f043f0d121a04a4da0d60e7791562f93c6aa91/pkg/cortex/cortex.go#L356-L360

If the resolver is not swapped (aka tenant federation disabled), Loki should just treat `x|y|z` as a single tenant named like that. 

I think we should consider enabling tenant federation default.

Reply:
Ok, so the multi tenant resolver will return an error when only one tenant ID is expected but multiple are set (https://github.com/grafana/loki/blob/main/pkg/tenant/resolver.go#L117). This is exactly the behaviour we want. E.g. if a users sets multiple tenants during ingestion we want Loki to return an error. Ideally `ErrTooManyOrgIDs` is treated as a user error though.